Python/pywarpx/WarpX.py

```diff
@@ -101,7 +101,13 @@ def write_inputs(self, filename='inputs', **kw):
 
         with open(filename, 'w') as ff:
 
+            prefix_old = ''
             for arg in argv:
+                dot_found = (arg.find('.') >= 0)
+                prefix_new = arg[0:arg.find('.')] if dot_found else arg[0:arg.find('=')].rstrip()
```

Comment:
These two lines are here to make sure that parameters such as `algo.particle_shape = 3` are preceeded by the header `# algo`, while parameters that are missing a prefix, such as `max_step = 100`, are preceeded by the header `# max_step`. There's probably smarter ways to do this, overall.

Reply:
Adding the headers is a good idea since it does visually separate the sections.

An alternative is to do `prefix_new = re.split(' |\.', arg)[0]`.

Reply:
Thank you for the suggestion, @dpgrote. That's more compact, I committed the change in https://github.com/ECP-WarpX/WarpX/pull/3093/commits/d913c0b120246cc135b998acbad2cafcc8056e9d.